sqlmesh/core/macros.py

```diff
@@ -114,6 +115,7 @@ def __init__(
         schema: t.Optional[t.Dict[str, t.Any]] = None,
         runtime_stage: RuntimeStage = RuntimeStage.LOADING,
         resolve_tables: t.Optional[t.Callable[[exp.Expression], exp.Expression]] = None,
+        snapshots: t.Optional[t.Dict[str, Snapshot]] = None,
```

Comment:
this will conflict with @eakmanrq work

Reply:
> Conclusion: I will update PR to have users go through a method to get the snapshot instead of calling the dict directly. When Ryan rebases this change into his PR he will update the snapshots dict to be fqn -> snapshot and then normalize within the method what the user passes in in order to get the normalized fqn to get the snapshot from the dict.